clients/src/main/resources/message/AddOffsetsToTxnRequest.json

```diff
@@ -0,0 +1,32 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+{
+  "apiKey": 25,
```

Comment:
I like the flexibility of being able to specify things that aren't requests or responses.  One example is the request header and response header.  There may be other stuff in the future.

Also, short files tend to be easier to follow than long files, for me at least.

Also, it's not really that redundant, is it?  The requests and responses don't really share much at all beyond the apiKey.  Well... I guess validVersions is the same too.  But the contents of each version will be totally different.

We could get clever with path names to avoid specifying "name" here, but it's probably not the right thing to do.  Remember there are still systems out there where filesystems are case insensistive and the FS can't tell the difference between `addoffsetstotxn.java` and `ADDOFFSETSTOTXN.JAVA` :disappointed:

Reply:
There is common stuff that will be added in the future, I think.  One example is some kind of definition of error code names-- that would go in its own file, and not be a request or a response.